### DISCUSSION TITLE
Are Timers identical to the ATMEGA328p?

### DISCUSSION THREAD

**Discussion by FrightRisk** [dbuezas/lgt8fx#117] [Q&A]
I'm sure that I will run into other things, but I need to switch 2 pins (usually pins 12 and 13), but could be 10 and 5 on an Uno with a Timer interrupt at between 8 and 10kHz. I will set the timer for 58us. Will the same code with perhaps a different prescaler work? Also, we are up against the very limit of what a 2k memory Uno can do, does this board use any more or less memory than a regular Uno? The Uno WiFi for example takes memory to run the virtual USB port which makes it unsuitable for our use. I'm trying to figure out what other surprises this board has. No EEPROM I guess is another.

**Comment by SuperUserNameMan** [comment]
@FrightRisk : I can't reply to your other questions, but regarding the EEPROM, it is emulated using the main Flash. [Here you'll find more info](https://github.com/dbuezas/lgt8fx/wiki/EEPROM-of-the-LGT8F328p-:-the-important-things-you-need-to-know)

**Comment by jayzakk** [comment]
Timers are compatible (lgt has an additional one).
"Compatible" also applies to the same amount of RAM, sorry :/ I wish they would have put in more of it...

**Comment by gadefox** [comment]
I'm using timer 0 and 1 (basic stuff) e.g.
```
  /* set up Timer0 */
  TCCR0A = 0;
  TIFR0 = (1 << TOV0);
  TIMSK0 = (1 << TOIE0);

  /* set up Timer1 */
  TCCR1A = 0;
  TIFR1 = (1 << OCF1A);
  TIMSK1 = (1 << OCIE1A);
```
Do I need to check/change anything, or are timers 0 and 1 fully compatible with `ATmega`? Thanks.

**Comment by hmeijdam** [comment]
[Here](https://dbuezas.github.io/arduino-web-timers/#mcu=ATMEGA328P&timer=0) you can check all their features and compare compatibility

<img width="1170" height="1191" alt="image" src="https://github.com/user-attachments/assets/091a421d-4f6e-49b7-8644-a1dc4ce8df85" />
